test/BlockChainHelper.h

```diff
@@ -0,0 +1,126 @@
+#include "JsonSpiritHeaders.h"
+#include <libethereum/BlockChain.h>
+#include <libethereum/TransactionQueue.h>
+#include <libdevcore/TransientDirectory.h>
+#include <libethcore/Ethash.h>
+
+using namespace std;
+using namespace json_spirit;
+using namespace dev;
+using namespace dev::eth;
+
+
+namespace dev {  namespace test {
+
+struct BlockStateUndefined : virtual Exception {};
+class TestTransaction;
+class TestBlock;
+class TestBlockChain;
+
+enum class RecalcBlockHeader
+{
+	Update,				//find new valid nonce and hash
+	Verify,				//check that block content is matching header
+	UpdateAndVerify,
+	SkipVerify
+};
+
+class TestTransaction
+{
+public:
+	TestTransaction(json_spirit::mObject const& _o);
+	TestTransaction(Transaction const& _tr) : m_transaction(_tr) {}
+	Transaction const& getTransaction() const { return m_transaction; }
+private:
+	json_spirit::mObject m_jsonTransaction;
+	Transaction m_transaction;
+};
+
+class TestBlock
+{
+private:
+	typedef Ethash::BlockHeader BlockHeader;
+
+public:
+	TestBlock() {};
+	TestBlock(TestBlock const& _original);	
+	TestBlock(std::string const& _blockRlp);
+	TestBlock& operator = (TestBlock const& _original);
+	TestBlock(mObject const& _blockObj, mObject const& _stateObj, RecalcBlockHeader _verify);
+	void addTransaction(TestTransaction const& _tr);
+	void addUncle(TestBlock const& _uncle);
+	void setUncles(vector<TestBlock> const& _uncles);
+	void mine(TestBlockChain const& _bc);
+
+	void setBlockHeader(Ethash::BlockHeader const& _header, RecalcBlockHeader _recalculate);
+	void setState(State const& _state);
+	void clearState();
+
+	bytes const& getBytes() const { return m_bytes; }
+	State const& getState() const { if (m_state.get() == 0) BOOST_THROW_EXCEPTION(BlockStateUndefined() << errinfo_comment("Block State is Nulled")); return *m_state.get(); }
+	BlockHeader const& getBlockHeader() const { return m_blockHeader;}
+	TransactionQueue const& getTransactionQueue() const { return m_transactionQueue; }
+	TransactionQueue & getTransactionQueue() { return m_transactionQueue; }
+	vector<TestTransaction> const& getTestTransactions() const { return m_testTransactions; }
+	vector<TestBlock> const& getUncles() const { return m_uncles; }
+	Address const& getBeneficiary() const { return m_blockHeader.beneficiary(); }
+
+private:
+	BlockHeader constructBlock(mObject const& _o, h256 const& _stateRoot);
+	bytes createBlockRLPFromFields(mObject const& _tObj, h256 const& _stateRoot = h256{});
+	void recalcBlockHeaderBytes(RecalcBlockHeader _recalculate);
+	void copyStateFrom(State const& _state);
+	void populateFrom(TestBlock const& _original);
+
+	BlockHeader m_blockHeader;
+	vector<TestBlock> m_uncles;
+	std::unique_ptr<State> m_state;
+	TransactionQueue m_transactionQueue;
+	BlockQueue m_uncleQueue;
+	bytes m_bytes;
+	std::unique_ptr<TransientDirectory> m_tempDirState;
+	vector<TestTransaction> m_testTransactions;
+};
+
+class TestBlockChain
+{
+private:
+	typedef FullBlockChain<Ethash> FullBlockChainEthash;
```

Comment:
Tthis can be put under the other privat section. There should be only on per class.